### PR TITLE
Handling of requestStake forking

### LINKED ISSUE #167: Handle requestStake transaction forking issue
Forking issue is observed on goerli quite frequently for stakeRequest transaction. This is causing acceptStakeRequest transactions to fail.

**Transaction History**

**StakeRequest First Transaction** https://goerli.etherscan.io/tx/0x78df824e72837a0f5793656072b1e6a8730398aefbdac3ebdb71f5266ec96545
**Mined Block**
https://goerli.etherscan.io/block/1089455

**StakeRequest Second Transaction** https://goerli.etherscan.io/tx/0x78df824e72837a0f5793656072b1e6a8730398aefbdac3ebdb71f5266ec96545
**Mined Block with failure of acceptStakeRequest and StakeRequest transaction**
https://goerli.etherscan.io/txs?block=1089456

### PR DESCRIPTION
**Pull request by abhayks1**
Forking issue is observed on goerli quite frequently for stakeRequest transaction. This is causing acceptStakeRequest transactions to fail.

**Transaction History**

**StakeRequest First Transaction** https://goerli.etherscan.io/tx/0x78df824e72837a0f5793656072b1e6a8730398aefbdac3ebdb71f5266ec96545
**Mined Block**
https://goerli.etherscan.io/block/1089455

**StakeRequest Second Transaction** https://goerli.etherscan.io/tx/0x78df824e72837a0f5793656072b1e6a8730398aefbdac3ebdb71f5266ec96545
**Mined Block with failure of acceptStakeRequest and StakeRequest transaction**
https://goerli.etherscan.io/txs?block=1089456

Fixes #167

### CODE REVIEW

**Comment by abhayks1** [comment]
Ready for review 🚀

File: src/models/StakeRequest.ts [hunk 1]
```diff
@@ -39,6 +39,8 @@ export default class StakeRequest extends Comparable<StakeRequest> {
 
   public stakerProxy?: string;
 
+  public blockNumber?: BigNumber;
```
Comment (sarvesh-ost) [inline src/models/StakeRequest.ts]: why this is an optional parameter? It shouldn't be, in my opinion.

File: src/models/StakeRequest.ts [hunk 2]
```diff
@@ -55,6 +57,7 @@ export default class StakeRequest extends Comparable<StakeRequest> {
     gateway?: string,
     staker?: string,
     stakerProxy?: string,
+    blockNumber?: BigNumber,
```
Comment (sarvesh-ost) [inline src/models/StakeRequest.ts]: similar comment.

File: src/repositories/StakeRequestRepository.ts [hunk 3]
```diff
@@ -128,6 +130,13 @@ export default class StakeRequestRepository extends Subject<StakeRequest> {
           type: DataTypes.STRING,
           allowNull: false,
         },
+        blockNumber: {
```
Comment (sarvesh-ost) [inline src/repositories/StakeRequestRepository.ts]: this shouldn't be null. Every stake request will have a block number.

Comment (abhayks1) [reply src/repositories/StakeRequestRepository.ts]: Changes done!

File: src/models/StakeRequest.ts (hunk 1 above)
Comment (abhayks1) [reply src/models/StakeRequest.ts]: Changes done!

File: src/models/StakeRequest.ts (hunk 2 above)
Comment (abhayks1) [reply src/models/StakeRequest.ts]: Done!

**Comment by abhayks1** [comment]
Available for review again 🍵

**Pull request merged**